alchemy/src/cloudflare/openauth.ts

```diff
@@ -0,0 +1,450 @@
+import { type Resource, ResourceKind } from "../resource.ts";
+import type { Secret } from "../secret.ts";
+import type { CloudflareApiOptions } from "./api.ts";
+import type { Bindings } from "./bindings.ts";
+import type { KVNamespaceResource } from "./kv-namespace.ts";
+import { Worker, type FetchWorkerProps } from "./worker.ts";
+
+/**
+ * Configuration for an OAuth provider
+ */
+export interface OAuthProviderConfig {
+  /**
+   * OAuth client ID
+   */
+  clientId: Secret;
+
+  /**
+   * OAuth client secret
+   */
+  clientSecret: Secret;
+
+  /**
+   * OAuth scopes to request
+   */
+  scopes?: string[];
+}
+
+/**
+ * Properties for creating an OpenAuth Worker
+ */
+export interface OpenAuthProps<B extends Bindings = Bindings>
+  extends CloudflareApiOptions {
```

Comment:
Should this interface extend Omit<WorkerProps<B>, ".."> since it has a lot of overlap?